### PR TITLE
Merged PR's for tests and (f)statfs

### PR DESCRIPTION
**Pull request by TristanB22**

### CODE REVIEW

**Comment by TristanB22** [comment]
If someone could please take a look at this, that would be awesome. It should be ready for merging with main.

File: src/safeposix/dispatcher.rs [hunk 1]
```diff
@@ -185,12 +192,6 @@ pub extern "C" fn dispatcher(cageid: u64, callnum: i32, arg1: Arg, arg2: Arg, ar
         FCNTL_SYSCALL => {
             cage.fcntl_syscall(unsafe{arg1.dispatch_int}, unsafe{arg2.dispatch_int}, unsafe{arg3.dispatch_int})
         }
-        DUP_SYSCALL => {
-            cage.dup_syscall(unsafe{arg1.dispatch_int}, None)
-        }
-        DUP2_SYSCALL => {
-            cage.dup2_syscall(unsafe{arg1.dispatch_int}, unsafe{arg2.dispatch_int})
-        }
```
Comment (jesings) [inline src/safeposix/dispatcher.rs:193]: Why'd you remove this?

File: src/safeposix/syscalls/fs_calls.rs [hunk 2]
```diff
@@ -206,20 +206,17 @@ impl Cage {
             (None, Some(pardirinode)) => {
                 let filename = truepath.file_name().unwrap().to_str().unwrap().to_string(); //for now we assume this is sane, but maybe this should be checked later
 
-                let effective_mode = S_IFREG as u32 | mode;
-
-                //assert sane mode bits
+                //assert same mode bits
```
Comment (jesings) [inline src/safeposix/syscalls/fs_calls.rs:209]: Sane was not a typo here, we're asserting that the mode bits make sense

File: src/safeposix/syscalls/fs_constants.rs [hunk 3]
```diff
@@ -76,14 +76,14 @@ pub const F_GETLEASE: i32 = 1025;
...
-pub const S_IFIFO: i32 = 0o10000;
-pub const S_IFLNK: i32 = 0o120000;
-pub const S_IFREG: i32 = 0o100000;
-pub const S_IFSOCK: i32 = 0o140000;
-pub const S_FILETYPEFLAGS: i32 = 0o170000;
+pub const S_IFIFO: i32 = 0o10000;   
+pub const S_IFLNK: i32 = 0o120000;  
+pub const S_IFREG: i32 = 0o100000;  
+pub const S_IFSOCK: i32 = 0o140000;        
+pub const S_FILETYPEFLAGS: i32 = 0o170000;
```
Comment (jesings) [inline src/safeposix/syscalls/fs_constants.rs:86]: You added a bunch of trailing spaces here, it's not a huge deal but if you can remove them in your commit that would be good

**Review by jesings — COMMENTED** [review]
Aside from my minor comments, this looks great!

File: src/safeposix/dispatcher.rs (hunk 1 above)
Comment (TristanB22) [reply src/safeposix/dispatcher.rs:193]: These are already there earlier in the dispatcher. Not sure why they are in there twice so I deleted them. If you look above these two a little bit, the exact same thing is there. Is there supposed to be two sets of dup's?

Comment (jesings) [reply src/safeposix/dispatcher.rs:193]: Nope, you're good

File: src/safeposix/syscalls/fs_calls.rs (hunk 2 above)
Comment (TristanB22) [reply src/safeposix/syscalls/fs_calls.rs:209]: Oopsie -- fixing that now

File: src/safeposix/syscalls/fs_constants.rs (hunk 3 above)
Comment (TristanB22) [reply src/safeposix/syscalls/fs_constants.rs:86]: On it!

**Review by rennergade — APPROVED** [review]
Almost looks good. Still seems there is a merge conflict? Maybe from Kaitlyn's PR that I just merged.

If you can fix that up that would be great, then I'll merge it in.

**Pull request merged**